# The program to answer the following questions

In [1]:
import pandas as pd

In [2]:
df = pd.read_excel("/content/2019 Winter Data Science Intern Challenge Data Set (1).xlsx")

In [3]:
df

order_id  shop_id  ...  payment_method              created_at
0            1       53  ...            cash 2017-03-13 12:36:56.190
1            2       92  ...            cash 2017-03-03 17:38:51.999
2            3       44  ...            cash 2017-03-14 04:23:55.595
3            4       18  ...     credit_card 2017-03-26 12:43:36.649
4            5       18  ...     credit_card 2017-03-01 04:35:10.773
...        ...      ...  ...             ...                     ...
4995      4996       73  ...           debit 2017-03-30 13:47:16.597
4996      4997       48  ...            cash 2017-03-16 20:36:16.389
4997      4998       56  ...            cash 2017-03-19 05:42:42.228
4998      4999       60  ...     credit_card 2017-03-16 14:51:18.188
4999      5000       44  ...           debit 2017-03-18 15:48:18.205

[5000 rows x 7 columns]

In [4]:
total_revenue = df['order_amount'].sum()
print("total revenue = ",total_revenue)
number_of_order = df['total_items'].sum()
print("total number of orders = ",number_of_order)
aov = total_revenue/number_of_order
print("AOV = ",aov)

total revenue =  15725640
total number of orders =  43936
AOV =  357.92152221412965


In [5]:
df1 =  pd.DataFrame(df , columns=["order_amount","total_items","payment_method"] )
df2 = df1.groupby(["payment_method"]).sum()
df2['aov'] = df2["order_amount"]/df2["total_items"]
df2

order_amount  total_items         aov
payment_method                                       
cash                 1164183         3130  371.943450
credit_card         12945867        37415  346.007403
debit                1615590         3391  476.434680

In [6]:
df2["aov"].sum()

1194.385533962436

### Question a)  Think about what could be going wrong with our calculation. Think about a better way to evaluate this data. 



1.   There must have been an error in calculating the total number of orders mostly the total number of orders is very less then the actual number. 
2.   The other reason is that the total order 
amount sum is more than the actual amount.







### Question b) What metric would you report for this dataset?

I would suggest the Average Order Value Metric because The average order value is a comparison of the number of orders to the total revenue. This is important because it provides insight into how much contacts are spending, on average, each time an order is placed.

### Question c) What is its value?
The value of AOV metric is: 357.921522214